apis/core.oam.dev/common/types.go

```diff
@@ -112,6 +112,22 @@ type WorkloadGVK struct {
 	Kind       string `json:"kind"`
 }
 
+// DisruptiveProperty describes whether when a trait is applied to an Application,
+// it will cause the pod to restart due to changes in some fields.
+// +kubebuilder:validation:Enum=unknown;yes;no;
+type DisruptiveProperty string
+
+const (
+	// UnknownDisruptive means disruptive unknown.
+	UnknownDisruptive DisruptiveProperty = "unknown"
+
+	// Disruptive means will cause the pod to restart.
+	Disruptive DisruptiveProperty = "yes"
+
+	// NonDisruptive means will not cause the pod restart
+	NonDisruptive DisruptiveProperty = "no"
```

Comment:
Looks like `true` and `false` are much more popular in CRD. I can get that you wanna set three values.
How about `unknown` meaning `false`? This is also the default behavior for those traits without setting it.

Reply:
It seems `unknown` is necessary as we could not tell whether a trait will restart its pods or not. But whether a trait will cause its Pods to restart is so essential to business, we should emphasizing platform team to clearly set the field, so leave it to default `false`. `unknown` should not be an option.

Reply:
after a deep thought, I think yes, we should use bool for this field.